internal/lockclient/client.go

```diff
@@ -19,6 +30,22 @@ type Client interface {
 	// implementation interacts with the underlying server and
 	// provides the service.
 	Release(lockservice.Descriptors) error
+	// Watch can be used to watch the state of lock on a descriptor
+	// continously. When the state of the lock changes, the "watcher"
+	// will be notified about the change.
+	// The channel passed as the argument can be used to stop watching
+	// at any point of time.
+	watch(lockservice.ObjectDescriptor, chan struct{}) (chan Lock, error)
```

Comment:
If watch and pounce are dormant in this PR, should we remove them from the interface like we discussed yesterday?